### PR TITLE
[RSDK-5622] Remove "i2cs" and "spis" sections from board configs

### PR DESCRIPTION
**Pull request by penguinland**
These are no longer used! Time to remove them.

A question whose answer I don't know: do we need to make changes to some kind of .vue file to remove the easy/GUI builder for these sections of the board configs? Maybe it does introspection and only has those parts for configs that actually have them in the first place. If I need to make changes over there, any idea what files they'd be in?

Feel free to review the first half dozen commits individually if that helps break things up, but I recommend _not_ going commit-by-commit towards the end, when I was fighting with the tests for the TMC stepper motor. What eventually happened with that was that I made a new mock SPI handle that had functions similar to the originals, and injected that in. but there were a bunch of wrong turns along the way.

### CODE REVIEW

**Comment by penguinland** [comment]
I've tried asking Bucket for reviewers twice, and both times it has given me someone on vacation. So, I'll tag @susmitaSanyal instead.

File: components/motor/tmcstepper/stepper_motor_tmc_test.go [hunk 1]
```diff
@@ -10,36 +10,60 @@ import (
...
+func (h *fakeSpiHandle) Xfer(
+	ctx context.Context,
+	baud uint,
+	chipSelect string,
+	mode uint,
+	tx []byte,
+) ([]byte, error) {
+	test.That(h.tb, tx, test.ShouldResemble, h.tx[h.i])
+	result := h.rx[h.i]
+	h.i += 1
```
Comment (susmitaSanyal) [inline components/motor/tmcstepper/stepper_motor_tmc_test.go]: What's the point of increasing i, if it's not being returned?

File: components/motor/tmcstepper/stepper_motor_tmc_test.go [hunk 2]
```diff
@@ -10,36 +10,60 @@ import (
...
 )
 
-// check is essentially test.That with tb.Error instead of tb.Fatal (Fatal exits and leaves fake.SPI stuck waiting).
-func check(tb testing.TB, actual interface{}, assert func(actual interface{}, expected ...interface{}) string, expected ...interface{}) {
-	tb.Helper()
-	if result := assert(actual, expected...); result != "" {
-		tb.Error(result)
-	}
+type fakeSpiHandle struct {
+	tx, rx [][]byte // Must have the same length
```
Comment (susmitaSanyal) [inline components/motor/tmcstepper/stepper_motor_tmc_test.go:21]: perhaps a better comment is `"tx and rx must have the same length" `

File: components/board/fake/board.go [hunk 3]
```diff
@@ -299,119 +248,6 @@ func (b *Board) Close(ctx context.Context) error {
...
-func (h *I2CHandle) WriteBlockData(ctx context.Context, register byte, data []byte) error {
-	return errors.New("finish me")
-}
-
-// Close releases access to the bus.
-func (h *I2CHandle) Close() error {
-	h.bus.mu.Unlock()
-	return nil
-}
-
```
Comment (susmitaSanyal) [inline components/board/fake/board.go:414]: Fake board looks so clean! 🤩

File: components/board/genericlinux/config.go [hunk 4]
```diff
@@ -46,13 +34,11 @@ func (conf *Config) Validate(path string) ([]string, error) {
...
 // configured. It is a union of the configs for the customlinux boards and the genericlinux boards
 // with static pin definitions, because those components all use the same underlying code but have
-// different config types (e.g., only genericlinux has named I2C and SPI buses, while only
-// customlinux can change its pin definitions during reconfiguration). The LinuxBoardConfig struct
-// is a unification of the two of them. Whenever we go through reconfiguration, we convert the
-// provided config into this type, and then reconfigure based on this.
+// different config types (e.g., only customlinux can change its pin definitions during
+// reconfiguration). The LinuxBoardConfig struct is a unification of the two of them. Whenever we
+// go through reconfiguration, we convert the provided config into this type, and then reconfigure
+// based on this.
```
Comment (susmitaSanyal) [inline components/board/genericlinux/config.go:40]: Can you give more context around the two "this" in this sentence? "we convert the provided config into this (What is "this" here?) type, and then reconfigure based on this (what is "this" here?)."

**Review by susmitaSanyal — COMMENTED** [review]
So far so good, just a few nits.

**Comment by susmitaSanyal** [comment]
@penguinland I believe there's no more .vue, front-end migrated to svelte. You may wanna touch base with fleet on if you should / how to modify those files. It will be under `web/src/components/...`

File: components/board/genericlinux/config.go (hunk 4 above)
Comment (penguinland) [reply components/board/genericlinux/config.go]: "This" is the `LinuxBoardConfig` defined on the very next line. I've rephrased the comment.

File: components/motor/tmcstepper/stepper_motor_tmc_test.go [hunk 5]
```diff
@@ -10,47 +10,79 @@ import (
...
+	mode uint,
+	tx []byte,
+) ([]byte, error) {
+	test.That(h.tb, tx, test.ShouldResemble, h.tx[h.i])
+	result := h.rx[h.i]
+	h.i++
+	return result, nil
+}
+
+func (h *fakeSpiHandle) Close() error {
```
Comment (susmitaSanyal) [inline components/motor/tmcstepper/stepper_motor_tmc_test.go:48]: question: Is this ever called? If not maybe we should remove it.

**Review by susmitaSanyal — APPROVED** [review]
LGTM! Just one question.

File: components/board/genericlinux/config.go [hunk 6]
```diff
@@ -11,25 +11,13 @@ import (
 
 // A Config describes the configuration of a board and all of its connected parts.
 type Config struct {
-	I2Cs              []board.I2CConfig                   `json:"i2cs,omitempty"`
-	SPIs              []board.SPIConfig                   `json:"spis,omitempty"`
 	AnalogReaders     []mcp3008helper.MCP3008AnalogConfig `json:"analogs,omitempty"`
 	DigitalInterrupts []board.DigitalInterruptConfig      `json:"digital_interrupts,omitempty"`
 	Attributes        utils.AttributeMap                  `json:"attributes,omitempty"`
```
Comment (randhid) [inline components/board/genericlinux/config.go:16]: if attributes is unused, we may as well remove it too.

Comment (penguinland) [reply components/board/genericlinux/config.go:16]: Okay, this is fixed in https://github.com/viamrobotics/rdk/pull/3264

**Pull request merged**